twill-common/src/main/java/org/apache/twill/filesystem/LocalLocation.java

```diff
@@ -98,8 +98,8 @@ public OutputStream getOutputStream() throws IOException {
 
   @Override
   public OutputStream getOutputStream(String permission) throws IOException {
-    ensureDirectory(file.getParentFile());
     Set<PosixFilePermission> permissions = parsePermissions(permission);
+    ensureDirectory(file.getParentFile(), permissions);
```

Comment:
yes, the intention of permissions is that others get access through, for example, group permission. So if I create the file with 700 but the directories are created with the default umask of 750, then that is not great. Even worse, if I create the file with 755, and the directories are created with 750, then world cannot access the file despite the intention.